source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+
+#. Copy the public key to all worker nodes.
+
+   .. code-block:: bash
+      
+      cd ~/.ssh
+      cp -v id_rsa.pub authorized_keys
+      chmod 600 authorized_keys
+      ssh-copy-id <worker>
+
+#. Verify you can SSH into each worker node without typing a password.
+
+#. Switch to root user.  
+   
+   .. code-block:: bash
+
+      sudo su
+
+#. Generate an SSH authentication key pair for root user.
+
+   .. code-block:: bash
+
+      ssh-keygen -t rsa
+
+   Accept all default options.
+
+   Example output:
+
+   .. code-block:: console
+
+      Generating public/private rsa key pair.
+      Enter file in which to save the key (/root/.ssh/id_rsa): 
+      Enter passphrase (empty for no passphrase): 
+      Enter same passphrase again: 
+      Your identification has been saved in /root/.ssh/id_rsa
+      Your public key has been saved in /root/.ssh/id_rsa.pub
+      The key fingerprint is:
+      SHA256:494NfaQ2R0H+aIHzY4PT3PQnbDwn/BUUjjFHnPNAz6k root@hpc-controller
+      The key's randomart image is:
+      +---[RSA 3072]----+
+      |             +==o|
+      |             +O*.|
+      |            o.+=*|
+      |             X.B+|
+      |        S   oE^ B|
+      |       . . . O Oo|
+      |        . . = o .|
+      |       . . + +   |
+      |        . . .    |
+      +----[SHA256]-----+
+
+#. Copy the root's public key to all worker nodes.
+
+   .. code-block:: bash
+
+      cd /root/.ssh
+      cp -v id_rsa.pub authorized_keys
+      chmod 600 authorized_keys
+      ssh-copy-id root@<worker>
+
+#. Verify you can SSH into each worker node without typing a password.
+
+Create :file:`slurm.conf` configuration file and copy to all nodes
+******************************************************************
+
+On the controller, create a new :file:`slurm.conf` configuration file 
+that contains general settings, each node's hardware resource information, 
+grouping of nodes into different partitions, and scheduling settings for 
+each partition.  This file will be copied to all nodes in the cluster.  
+
+#. Create a base :file:`slurm.conf` configuration file.
+
+   .. code-block:: bash
+
+      sudo mkdir -p /etc/slurm
+      sudo cp -v /usr/share/defaults/slurm/slurm.conf /etc/slurm
+
+#. Add the controller information.
+
+   a. :command:`sudoedit` the :file:`slurm.conf` file. Set the `ControlMachine`
+      value to the controller's hostname and set the `ControlAddr` to its 
+      IP address. 
+
+      For example:
+
+      .. code-block:: console
+
+	 ControlMachine=hpc-controller
+	 ControlAddr=192.168.1.10
+
+   #. Save and exit.
+
+#. Add the worker nodes information.
+   
+   a. Create a file containing a list of the worker nodes.
+
+      .. code-block:: bash
+
+         cat > worker-nodes-list << EOF
+	 hpc-worker1
+	 hpc-worker2
+	 hpc-worker3
+	 hpc-worker4
+	 EOF
+
+   #. Using PSDH, get the hardware configuration of each node.
+
+      .. code-block:: bash
+
+         pdsh -w ^worker-nodes-list slurmd -C
+
```

Comment:
Just an observation: pdsh uses ssh by default and accesses the workers by hostnames here, no worker IP address needed. (See comments below)